### PR TITLE
Escape the default "Thank you" text instead of the filtered message

### LINKED ISSUE #40329: BC issue with woocommerce_thankyou_order_received_text
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

In https://github.com/woocommerce/woocommerce/pull/39758 we introduced a new template for the order-received note: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/templates/checkout/order-received.php

The new template escapes now the output of the `woocommerce_thankyou_order_received_text` filter, which was not escaped before: https://github.com/woocommerce/woocommerce/pull/39758/files#diff-564a2935633e7fbf95ba8f1cae13c198399275de26dfb1c994868b6e9a3437c1L42

This introduces a bc problems for integrations using this filter and returning HTML.

### Expected behavior

The output should not be escaped. This would follow also how we treat other filters in the templates, for example here:
https://github.com/woocommerce/woocommerce/blob/7f9322c85858a17f6e9e9e79eaa3aefeda1a0624/plugins/woocommerce/templates/myaccount/my-orders.php#L37

or in this template https://github.com/woocommerce/woocommerce/blob/7f9322c85858a17f6e9e9e79eaa3aefeda1a0624/plugins/woocommerce/templates/myaccount/my-downloads.php

### Actual behavior

The output gets escaped by using `esc_html()`, which converts HTML tags into text.

### Steps to reproduce

1. Write and activate a small plugin like `add_filter('woocommerce_thankyou_order_received_text', function(){ return '<p><strong>Thank you!</strong></p>'; });
2. Create an order
3. See the escaped output like `<p><strong>Thank you!</strong></p>`

### WordPress Environment

WC Version: 8.1.0

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by YordanSoares**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #40329 and #40352

### How to test the changes in this Pull Request:

1. Activate this code snippet in your site:

```php
add_filter('woocommerce_thankyou_order_received_text', 'wc_test_thank_you_link', 10, 2);
function wc_test_thank_you_link( $text, $order ) {
    $text .= '<p><a href="https://example.com">Click here</a></p>';
    return $text;
}
```
2. Place a new order and check the code displayed after "Thank you. Your order has been received.", which should look like this:

image

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Escape the default 'thank you' text instead of the filtered message.

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/templates/checkout/order-received.php [hunk 1]
```diff
@@ -32,10 +32,10 @@
 	 */
 	$message = apply_filters(
 		'woocommerce_thankyou_order_received_text',
-		__( 'Thank you. Your order has been received.', 'woocommerce' ),
+		esc_html( __( 'Thank you. Your order has been received.', 'woocommerce' ) ),
 		$order
 	);
 
-	echo esc_html( $message );
+	echo $message;
```
Comment (lsinger) [inline plugins/woocommerce/templates/checkout/order-received.php:40]: We'll need a linting directive before this line: 
```suggestion
	// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
	echo $message;
```

**Review by lsinger — CHANGES_REQUESTED** [review]
Thanks @YordanSoares for your contribution! I left a comment for a linting directive we'll need. Could you also please bump the version of that template file to 8.3.0? 

For the future to lint changes:

```
cd plugins/woocommerce
pnpm lint:php:changes
```

This would then at least detect the need for the directive.

**Comment by lsinger** [comment]
@YordanSoares It seems running the changelog adder doesn't currently work from forks. Could you run, from the root of the repository, the following to add a changelog file? 

```
pnpm --filter=./plugins/woocommerce run changelog add
```

**Comment by YordanSoares** [comment]
> @YordanSoares It seems running the changelog adder doesn't currently work from forks. Could you run, from the root of the repository, the following to add a changelog file?
> 
> ```
> pnpm --filter=./plugins/woocommerce run changelog add
> ```

@lsinger I tried to add this command, but it didn't work on my Windows environment:

image

I tried installing all the [prerequisites](https://github.com/woocommerce/woocommerce#prerequisites), but I haven't being able to run it with success, yet. Any ideas?

**Comment by YordanSoares** [comment]
@lsinger it seems that @github-actions finally managed to log the change :)

**Review by lsinger — APPROVED** [review]
Thank you for making those changes, @YordanSoares! Looks good to me. 

It seems the CI checks aren't running -- I'll try to close and re-open this PR to trigger them again.

**Pull request merged**